src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -0,0 +1,995 @@
+                verWrittenCur: 0x00010002, // Dropped sizeof(Float).
+                verReadableCur: 0x00010002,
+                verWeCanReadBack: 0x00010002,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(ValueMappingTransformer).Assembly.FullName);
+        }
+
+        public sealed class Column : OneToOneColumn
+        {
+            public static Column Parse(string str)
+            {
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public sealed class Arguments
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the column containing the keys", ShortName = "key")]
+            public string KeyColumn;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the column containing the values", ShortName = "value")]
+            public string ValueColumn;
+
+            [Argument(ArgumentType.Multiple, HelpText = "The data loader", NullName = "<Auto>", SignatureType = typeof(SignatureDataLoader))]
+            public IComponentFactory<IMultiStreamSource, IDataLoader> Loader;
+
+            [Argument(ArgumentType.AtMostOnce, IsInputFileName = true, HelpText = "The data file containing the terms", ShortName = "data", SortOrder = 2)]
+            public string DataFile;
+
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Specifies whether the values are key values or numeric, only valid when loader is not specified and the type of data is not an idv.")]
+            public bool ValuesAsKeyType = true;
+        }
+
+        protected ValueMappingTransformer(IHostEnvironment env, IDataView lookupMap,
+            string keyColumn, string valueColumn, (string Input, string Output)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ValueMappingTransformer)), columns)
+        {
+            env.CheckNonEmpty(keyColumn, nameof(keyColumn), "A key column must be specified when passing in an IDataView for the value mapping");
+            env.CheckNonEmpty(valueColumn, nameof(valueColumn), "A value column must be specified when passing in an IDataView for the value mapping");
+            _valueMap = CreateValueMapFromDataView(lookupMap, keyColumn, valueColumn);
+            int valueColumnIdx = 0;
+            env.Assert(lookupMap.Schema.TryGetColumnIndex(valueColumn, out valueColumnIdx));
+            _valueMetadata = CopyMetadata(lookupMap.Schema[valueColumnIdx].Metadata);
```

Comment:
>CopyMetadata(lookupMap.Schema[valueColumnIdx].Metadata); [](start = 29, length = 56)

why exactly you need to copy metadata? why you can't use pointer?

Reply:
I followed what TermLookup was doing but we can just use the pointer. I have changed this to an assignment and deleted the CopyMetadata function. 

---
In reply to: [243465298](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243465298) [](ancestors = 243465298)